File: ScalaSpark/src/scala/JobManagment/Runner.scala
package JobManagment


import de.halcony.argparse.Parser

trait Runner {

  def register[T <: Job](jobName: String, jobClass: Class[T], argumentsParser: Parser): Unit

  def run(jobName: String, sparkMaster: String, sparkAppName: String, args: Array[String]): Unit

}

File: PythonSpark/JobManagment/Runner.py
from abc import ABC, abstractmethod
from typing import Type

from JobManagment.Job import Job


class Runner(ABC):
    @abstractmethod
    def register(self, job_name: str, job_class: Type[Job]):
        pass

    @abstractmethod
    def run(self, job_name: str, spark_master: str, spark_app_name: str, args: list):
        pass
